pkg/pillar/scripts/device-steps.sh

```diff
@@ -373,7 +373,7 @@ touch "$WATCHDOG_FILE/nim.touch"
 # receiving those messages on the console.
 mkfifo /run/diag.pipe
 (while true; do cat; done) < /run/diag.pipe >/dev/console 2>&1 &
-$BINDIR/diag -f -o /run/diag.pipe runAsService &
+$BINDIR/diag -f -o /run/diag.pipe &
```

Comment:
Didn't see this in the other PR.

The current master runs diag as a goroutine inside zedbox but here you are accidentally changing that by using inlineIfArgs.
For diag we need to run as a service even though it has args.

I don't know if we should also allow a user to run a second copy of diag inline (something I've done occasionally e.g., diag -f -o /dev/tty). Easiest would be to keep runAsService for diag since it wouldn't change any of that.

Reply:
I check args, not flags, so I not expect that `-f -o` will force to run inline instead of service.

Reply:
I checked and now I cannot run another instance of diag inline because of publication of types.MetricsMap already allocated and in use.

Reply:
> I check args, not flags, so I not expect that `-f -o` will force to run inline instead of service.

OK, that's good.